src/interaction/interactionmonitor.cpp

```diff
@@ -0,0 +1,85 @@
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <openspace/interaction/interactionmonitor.h>
+
+#include <openspace/engine/globals.h>
+#include <openspace/engine/windowdelegate.h>
+
+namespace {
+    constexpr const char* _loggerCat = "InteractionMonitor";
+
+    constexpr openspace::properties::Property::PropertyInfo IdleTimeInfo = {
+        "IdleTime",
+        "Idle Time",
+        "Time passed from latest registerad interaction until application goes idle."
+    };
+    constexpr openspace::properties::Property::PropertyInfo IsInActiveStateInfo = {
+        "IsInActiveStateInfo",
+        "Is State Active",
+        "Keeps track whether the interaction session is in active state or not. " 
+        "False if application is in idle state, true if it is in active state."
+    };
+} // namespace
+
+namespace openspace::interaction {
+
+    InteractionMonitor::InteractionMonitor()
+        : properties::PropertyOwner({ "InteractionMonitor" })
+        , _isInActiveState(IsInActiveStateInfo, false)
+        , _idleTime( properties::DoubleProperty(IdleTimeInfo, 120.0))
+    {
+        addProperty(_isInActiveState);
+        addProperty(_idleTime);
+    }
+
+    void InteractionMonitor::setActivityState(bool isActive)
+    {
+        _isInActiveState.setValue(isActive);
+    }
+
+    void InteractionMonitor::setIdleTime(double time)
+    {
+        _idleTime.setValue(time);
+    }
+
+    void InteractionMonitor::updateActivityState()
+    {
+        double currentApplicationTime = global::windowDelegate.applicationTime();
+        double timeDiff = currentApplicationTime - _lastInteractionTime;
+
+        if (timeDiff >= _idleTime.value() && _isInActiveState) {
+            setActivityState(false);
+        }
+    }
+
+    void InteractionMonitor::registerInteraction(std::string interactionType)
```

Comment:
Better to use an enum type here instead of string, both for maintainability and performance.
However, since the feature of `interactionType` is not used for anything right now, I'd even suggest removing the parameter altogether until we have a concrete use case for it. Thoughts?

Reply:
I can remove it for now. I used it for debugging purposes earlier and thought  it could be useful but it does not serve any functionality at the moment.